cmd/image-builder/version.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"runtime/debug"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	version string
+	commit  string
+
+	imagesVersion string
+)
+
+func init() {
+	if version == "" {
+		version = "unknown"
+	}
+
+	if bi, ok := debug.ReadBuildInfo(); ok {
+		for _, bs := range bi.Settings {
+			switch bs.Key {
+			case "vcs.revision":
+				commit = bs.Value
+			}
+		}
+
+		for _, dep := range bi.Deps {
+			if dep.Path == "github.com/osbuild/images" {
+				imagesVersion = dep.Version
+			}
+		}
+	}
+
+	// If we didnt find any commit info, we set it to unknown
+	if commit == "" {
+		commit = "unknown"
+	}
+
+	// If we didn't find any dependency info, we set it to unknown
+	if imagesVersion == "" {
+		imagesVersion = "unknown"
+	}
+}
+
+type versionDescription struct {
+	ImageBuilder struct {
+		Version      string `yaml:"version"`
+		Commit       string `yaml:"commit"`
```

Comment:
maybe omitempty here? I'm not sure we always get this

Reply:
It's always set (see `init`) but it will be set to `unknown`. I feel that's probably more useful than omitting? It shows that things weren't build from a git checkout.